Make calls without specifying DEVICE_OPTIMIZATIONS break the build.

We need to add ```$(DEVICE_OPTIMIZATION)``` option to ```$(MAKE)``` calls to sanitize optimized builds and avoid overlaying with generic arm optimization. I've just added it to all calls, so may be we can avoid some of them.

diff Makefile.orig Makefile

```
diff --git a/Makefile b/Makefile
index 9bb7354..0319237 100644
--- a/Makefile
+++ b/Makefile
@@ -287,8 +287,8 @@ $(PACKAGES_STAGE1_TAR_FILE): $(PORTS_DIR) $(PKGMK_CONFIG_FILE) $(PRTGET_CONFIG_F
 #
 .PHONY: stage0
 stage0:
-       $(MAKE) build-stage0-packages PKGMK_FAKEROOT=yes
-       $(MAKE) build-stage0-rootfs
+       $(MAKE) build-stage0-packages PKGMK_FAKEROOT=yes DEVICE_OPTIMIZATION=$(DEVICE_OPTIMIZATION)
+       $(MAKE) build-stage0-rootfs DEVICE_OPTIMIZATION=$(DEVICE_OPTIMIZATION)

 #------------------------------------------------------------------------------
 # STAGE1
@@ -297,10 +297,10 @@ stage0:
 .PHONY: stage1
 stage1:
        @echo "[`date +'%F %T'`] Preparing chroot environment ($(ROOTFS_STAGE1_DIR))"
-       $(MAKE) prepare-stage1-rootfs
-       $(MAKE) download-stage1-sources
+       $(MAKE) prepare-stage1-rootfs DEVICE_OPTIMIZATION=$(DEVICE_OPTIMIZATION)
+       $(MAKE) download-stage1-sources DEVICE_OPTIMIZATION=$(DEVICE_OPTIMIZATION)
        @echo "[`date +'%F %T'`] Cleaning up before entering into chroot environment"
-       $(MAKE) clean-prtgetconf
+       $(MAKE) clean-prtgetconf DEVICE_OPTIMIZATION=$(DEVICE_OPTIMIZATION)
        @echo "[`date +'%F %T'`] Mounting /dev on $(ROOTFS_STAGE1_DIR)/dev"
        @mountpoint -q $(ROOTFS_STAGE1_DIR)/dev || sudo mount --bind /dev $(ROOTFS_STAGE1_DIR)/dev
        @echo "[`date +'%F %T'`] Mounting /proc on $(ROOTFS_STAGE1_DIR)/proc"
@@ -309,7 +309,7 @@ stage1:
        @mountpoint -q $(ROOTFS_STAGE1_DIR)/workspace || sudo mount --bind $(WORKSPACE_DIR) $(ROOTFS_STAGE1_DIR)/workspace
        @echo "[`date +'%F %T'`] Entering chroot enrivonment"
        @sudo chroot $(ROOTFS_STAGE1_DIR) /bin/bash --login -c \
-               "cd /workspace && $(MAKE) build-stage1-packages PKGMK_FORCE=yes WORKSPACE_DIR=/workspace || exit 0"
+               "cd /workspace && $(MAKE) build-stage1-packages PKGMK_FORCE=yes WORKSPACE_DIR=/workspace DEVICE_OPTIMIZATION=$(DEVICE_OPTIMIZATION) || exit 0"
        @echo "[`date +'%F %T'`] Exiting chroot enrivonment"
        @echo "[`date +'%F %T'`] Unmounting $(ROOTFS_STAGE1_DIR)/workspace"
        @sudo umount -f $(ROOTFS_STAGE1_DIR)/workspace
@@ -326,12 +326,12 @@ stage1:
 bootstrap:
        @echo "[`date +'%F %T'`] Bootstrap started"
        @echo "[`date +'%F %T'`] Running Stage 0"
-       $(MAKE) stage0 2>&1 | tee stage0.log
+       $(MAKE) stage0 DEVICE_OPTIMIZATION=$(DEVICE_OPTIMIZATION) 2>&1 | tee stage0.log
        @grep -e 'failed\.' -e 'succeeded\.' stage0.log
        @echo "[`date +'%F %T'`] Selecting $(ROOTFS_TAR_FILE) -> $(ROOTFS_STAGE0_TAR_FILE)"
        @ln -s $(ROOTFS_STAGE0_TAR_FILE) $(ROOTFS_TAR_FILE)
        @echo "[`date +'%F %T'`] Running Stage 1"
-       $(MAKE) stage1 2>&1 | tee stage1.log
+       $(MAKE) stage1 DEVICE_OPTIMIZATION=$(DEVICE_OPTIMIZATION) 2>&1 | tee stage1.log
        @grep -e 'failed\.' -e 'succeeded\.' stage1.log
        @echo "[`date +'%F %T'`] Bootstrap completed"
```

It isn't working as expected. On arm64 optimization there is still an error with collections. It keeps trying to get core-arm instead of core-arm64 overlay:

```
libgmp
[Config error: can't access /workspace/ports/core-arm]
[2022-10-31 00:11:56] Building port: /workspace/ports/core/libgmp
=======> Downloading 'https://gmplib.org/download/gmp/gmp-6.2.1.tar.xz'.
=======> ERROR: Command 'curl' not found.
make: *** [Makefile:274: /workspace/packages.stage1.tar.xz] Error 1
[2022-10-31 00:11:56] Exiting chroot enrivonment
[2022-10-31 00:11:56] Unmounting /home/pitillo/crux-arm-release_arm64/rootfs-stage1/workspace
[2022-10-31 00:11:56] Unmounting /home/pitillo/crux-arm-release_arm64/rootfs-stage1/proc
[2022-10-31 00:11:56] Unmounting /home/pitillo/crux-arm-release_arm64/rootfs-stage1/dev
make[1]: Leaving directory '/home/pitillo/crux-arm-release_arm64'
=======> Building '/workspace/ports/core/attr/attr#2.5.1-1.pkg.tar.xz' succeeded.
=======> Building '/workspace/ports/core/autoconf/autoconf#2.71-1.pkg.tar.xz' succeeded.
[2022-10-31 00:11:56] Bootstrap completed
```

After some tests, arm64 build still gives problems:

It tries to build next stage without optimizations.

Wrong command used:
```
 make DEVICE_OPTIMIZATION=arm64 bootstrap
```
Right one used still with problems:
```
DEVICE_OPTIMIZATION=arm64 make bootstrap
```

Log's last part:
```
[Config error: can't access /workspace/ports/core-arm]
package autoconf is installed
[Config error: can't access /workspace/ports/core-arm]
prt-get: updating /workspace/ports/core/autoconf
=======> Package '/workspace/ports/core/autoconf/autoconf#2.71-1.pkg.tar.xz' is up to date.
prt-get: reinstalling autoconf 2.71-1

-- Packages updated
autoconf

prt-get: updated successfully
libgmp
[Config error: can't access /workspace/ports/core-arm]
[2022-11-09 13:23:53] Building port: /workspace/ports/core/libgmp
=======> Downloading 'https://gmplib.org/download/gmp/gmp-6.2.1.tar.xz'.
=======> ERROR: Command 'curl' not found.
make: *** [Makefile:272: /workspace/packages.stage1.tar.xz] Error 1
[2022-11-09 13:23:53] Exiting chroot enrivonment
[2022-11-09 13:23:53] Unmounting /home/pitillo/crux-arm-release/rootfs-stage1/workspace
[2022-11-09 13:23:53] Unmounting /home/pitillo/crux-arm-release/rootfs-stage1/proc
[2022-11-09 13:23:53] Unmounting /home/pitillo/crux-arm-release/rootfs-stage1/dev
make[1]: Leaving directory '/home/pitillo/crux-arm-release'
=======> Building '/workspace/ports/core/attr/attr#2.5.1-1.pkg.tar.xz' succeeded.
=======> Building '/workspace/ports/core/autoconf/autoconf#2.71-1.pkg.tar.xz' succeeded.
[2022-11-09 13:23:53] Bootstrap completed
```

--- a/Makefile
+++ b/Makefile
@@ -56,6 +56,8 @@ DEVICE_OPTIMIZATION = arm
 endif
 # Load CFLAGS and COLLECTIONS for selected optimization
 include $(WORKSPACE_DIR)/devices/$(DEVICE_OPTIMIZATION).mk
+# Export variable to sub-make
+export DEVICE_OPTIMIZATION
 
 # Default build command
 PKGMK_CMD = pkgmk

